src/Formatter.js

```diff
@@ -65,6 +65,17 @@ class Formatter {
      */
     format(result) {
     }
+    /**
+     * Provide the information for the formatter as an object and
+     * return the formatted string that contains a lot of information
+     * according to the formatter's needs.
```

Comment:
This needs to describe that the formatOutput call is supposed to format the whole output, including headers, footers, and summaries, and formatted Result instances. If the formatOutput is defined, the linter will not call format directly and it is up to formatOutput to do that. If formatOutput is not defined in a Formatter plugin, then ilib-lint will use its own default headers and footers and will call format to format each Result.

Reply:
I update it. Please check it if it's ok.